components/create_comment/index.js

```diff
@@ -2,15 +2,88 @@
 // See License.txt for license information.
 
 import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {getBool} from 'mattermost-redux/selectors/entities/preferences';
+import {resetHistoryIndex} from 'mattermost-redux/actions/posts';
+import {Preferences, Posts} from 'mattermost-redux/constants';
+
+import {
+    clearCommentDraftUploads,
+    updateCommentDraft,
+    makeOnMoveHistoryIndex,
+    makeOnSubmit,
+    makeOnEditLatestPost
+} from 'actions/views/create_comment';
+
+import {makeGetCommentDraft} from 'selectors/rhs';
 
 import CreateComment from './create_comment.jsx';
 
 function mapStateToProps(state, ownProps) {
     const err = state.requests.posts.createPost.error || {};
+
+    const getCommentDraft = makeGetCommentDraft(ownProps.rootId);
+
+    const draft = getCommentDraft(state);
+
+    const enableAddButton = draft.message.trim().length !== 0 || draft.fileInfos.length !== 0;
+
     return {
-        ...ownProps,
+        draft,
+        enableAddButton,
+        ctrlSend: getBool(state, Preferences.CATEGORY_ADVANCED_SETTINGS, 'send_on_ctrl_enter'),
         createPostErrorId: err.server_error_id
     };
 }
 
-export default connect(mapStateToProps)(CreateComment);
+function makeOnUpdateCommentDraft(rootId) {
+    return (draft) => updateCommentDraft(rootId, draft);
+}
+
+function makeMapDispatchToProps() {
```

Comment:
I don't actually think we've done this elsewhere. We usually just pass in the action creators into the component, and the component calls it with whatever arguments it needs. This works though, and it shouldn't cause any unnecessary re-renders as far as I can tell.